### DISCUSSION TITLE
You are not allowed to reconfigure seb

### DISCUSSION THREAD

**Discussion by yagnonAWS** [SafeExamBrowser/seb-win-refactoring#887] [Q&A]
You are not allowed to reconfigure seb
After integrating my configuration file on my moodle platform, I'm in student mode to take the quiz, but I get the message you are not authorized to reconfigure seb after executing the configuration.
I'll leave you the configuration file and the screenshot.
My file should launch teams on startup
Remove the JPEG extension from the test file to open it in safe exam.
Please help me
![IMG_3929](https://github.com/SafeExamBrowser/seb-win-refactoring/assets/100287079/74402107-94f5-4fef-b214-cb203139e512)
![TEST 1 seb](https://github.com/SafeExamBrowser/seb-win-refactoring/assets/100287079/1c6d83bc-db4d-4b6a-a517-f6e69ba31a95)

**Comment by dbuechel** [comment]
You'll need to allow reconfiguration in your configuration file, see https://safeexambrowser.org/windows/win_usermanual_en.html#ExamPane > Reconfiguring Secure / Exam Session.

**Comment by yagnonAWS** [comment]
Hello Damain,
I've authorized the reconfiguration but still the same thing I can't click
on do test.


Le lun. 27 mai 2024 à 11:24, Damian Büchel ***@***.***> a
écrit :

**Reply by danschlet** [reply]
This error is displayed when the ConfigKey (CK) which SEB calculates and sends to Moodle doesn't match the CK which Moodle calculated. Moodle then re-sends the configuration file and SEB reports that you can't reconfigure. Allowing reconfiguring doesn't solve the underlaying issue.

**Comment by yagnonAWS** [comment]
I've authorized the reconfiguration but still the same thing I can't click
on do test.
Tu peux me partager un fichier fait avec la version 3.7 si tu en a
If you have a file made with version 3.7, please share it with me.

Le lun. 27 mai 2024 à 11:24, Damian Büchel ***@***.***> a
écrit :

**Comment by danschlet** [comment]
I think it might be the Teams permittedProcess, specifically the path with backslashes. @dbuechel wasn't there an issue with the ConfigKey not being calculated properly when there are some specific special characters in permittedProcesses?

**Reply by dbuechel** [reply]
Yes exactly, if I recall correctly there appears to be an issue with certain special characters in the command-line arguments of permitted applications.

**Comment by yagnonAWS** [comment]
But I've disabled the "Use Browser Exam Key and Config Key" option so that
I don't have to compare the keys.
But I still get the same error

**Reply by danschlet** [reply]
If you configured in your Moodle quiz settings to use SEB: "Require the use of Safe Exam Browser" "Yes – Configure manually" or "Yes – Upload my own config", then Moodle will add the required settings (e.g. "Use Browser Exam Key and Config Key" true) to those settings, you cannot override that. Same if you use "Yes – Use SEB client config" and add a key to Allowed browser exam keys. 

Trust me, it's a bug relating the calculation of the key. With the client config option and a browser exam key which you copy from the SEB Configuration Tool it will work.

**Comment by yagnonAWS** [comment]
I activated the "Use Browser Exam Key and Config Key" option and copied and
pasted the key, but I still get the same error message

**Comment by spicymoodles** [comment]
I'm having the same issue in Moodle 4.1 with Permitted Processes enabled. (with nginx as load balancer/reverse proxy)
No matter what key I use - either 'Browser Exam Key' or 'Config Key', I get this "You are not allowed to reconfigure seb" error. 

Choosing NOT to use keys in the Quiz settings doesn't help too ("if no keys are entered, then browser exam keys are not checked") since Moodle seems to force this setting. (is this the bug you refer to @danschlet ?) 

Would this be related to reverse proxy at all? as suggested here:
https://moodle.org/mod/forum/discuss.php?d=433062

I can't workout how to use the "Reconfiguring Secure/Exam Session" option.
https://safeexambrowser.org/windows/win_usermanual_en.html#ExamPane
How can I find the correct URL for this?

**Reply by dbuechel** [reply]
> How can I find the correct URL for this?

You do not need to use the URL field, you can also simply globally allow reconfiguration without checking the URL. With respect to the Moodle issue(s), I unfortunately cannot help you further there as I do not have much experience with Moodle resp. the necessary detail knowledge.

**Comment by yagnonAWS** [comment]
@dbuechel 
Hello I was able to remove the message you are not allowed to configure safe exam browser by unchecking the Download and open config SAB box in the Down/upload session, but I don't have the option to do the test.
it says "Configuration key or Browser key could not be validated.  Please ensure that you are using Safe Exam Browser with the correct configuration file." 
It's really the key calculation as  @danschlet said
But the problem persists when I copy the Browser Exam Key into the required field in the quiz settings on Moodle.
What do YOU think? Or at least what can I do? @dbuechel @danschlet 

![WhatsApp Image 2024-08-03 at 14 46 05](https://github.com/user-attachments/assets/40c68951-7e67-4667-8e86-099610ee6d17)

**Reply by dbuechel** [reply]
Please note that there is an issue in Moodle with configuration values of type `real` (see #905). As a workaround, you can manually edit the configuration and simply remove the respective values.
